pkg/controllers/provisioning/scheduling/nodeclaimtemplate.go

```diff
@@ -57,7 +58,10 @@ func NewNodeClaimTemplate(nodePool *v1.NodePool) *NodeClaimTemplate {
 		v1.NodePoolHashAnnotationKey:        nodePool.Hash(),
 		v1.NodePoolHashVersionAnnotationKey: v1.NodePoolHashVersion,
 	})
-	nct.Labels = lo.Assign(nct.Labels, map[string]string{v1.NodePoolLabelKey: nodePool.Name})
+	nct.Labels = lo.Assign(nct.Labels, map[string]string{
+		v1.NodePoolLabelKey: nodePool.Name,
+		nodeclaimutils.NodeClassLabelKey(nodePool.Spec.Template.Spec.NodeClassRef): nodePool.Spec.Template.Spec.NodeClassRef.Name,
```

Comment:
Would be kinda cool if we reconciled this on the NodePool too so that folks could list all NodePools that have a given NodeClass

Reply:
Maybe, though you can do that today with a field selector right? Is there additional value?